test/integration/operator_test.go

```diff
@@ -0,0 +1,231 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+//go:build integration
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/avast/retry-go/v4"
+	kubearchivev1alpha1 "github.com/kubearchive/kubearchive/cmd/operator/api/v1alpha1"
+	"github.com/kubearchive/kubearchive/test"
+	corev1 "k8s.io/api/core/v1"
+	errs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type Test int
+
+const (
+	emptyKAC Test = iota
+	nonEmptyKAC
+)
+
+const (
+	kacName     = "kubearchive"
+	a13eName    = kacName + "-a13e"
+	sinkName    = kacName + "-sink"
+	filtersName = "sink-filters"
+)
+
+func TestEmptyKAC(t *testing.T) {
+	runTest(t, emptyKAC, map[string]any{})
+}
+
+func TestNonEmptyKAC(t *testing.T) {
+	resources := map[string]any{
+		"resources": []map[string]any{
+			{
+				"selector": map[string]string{
+					"apiVersion": "v1",
+					"kind":       "Pod",
+				},
+				"archiveWhen": "true",
+				"deleteWhen":  "status.phase == 'Succeeded'",
+			},
+		},
+	}
+	runTest(t, nonEmptyKAC, resources)
+}
+
+func runTest(t *testing.T, testName Test, resources map[string]any) {
+	clientset, dynaclient, errClient := test.GetKubernetesClient()
```

Comment:
since runTest is a helper method I think the following change should be made

```suggestion
func runTest(t testing.TB, testName Test, resources map[string]any) {
	t.Helper()
	
	clientset, dynaclient, errClient := test.GetKubernetesClient()
```

`testing.TB` is an interface that is implemented by `*testing.T` and `*testing.B`. Doing this will make code reuse easier if we add benchmark tests in the future. Calling `t.Helper()` will make debugging unit tests easier, since the the test failure will be reported as the function that called `runTest`. This prevents potential confusion because both `TestEmptyKAC` and `TestNonEmptyKAC` call `runTest` so it may not be clear which one failed